test/cpp/end2end/xds/xds_pick_first_end2end_test.cc

```diff
@@ -94,23 +93,23 @@ TEST_P(PickFirstTest, PickFirstConfigurationIsPropagated) {
       ->mutable_typed_config()
       ->PackFrom(pick_first);
   balancer_->ads_service()->SetCdsResource(cluster);
-  bool saw_different_backend = false;
   size_t start_index = 0;
-  for (size_t i = 0; i < 100 && !saw_different_backend; ++i) {
+  for (size_t i = 0; i < 100; ++i) {
     // Update EDS resource.  This will send a new address list update to the LB
     // policy.
     balancer_->ads_service()->SetEdsResource(BuildEdsResource(
         EdsResourceArgs({{"locality0", CreateEndpointsForBackends(
                                            start_index, start_index + 3)}})));
     auto result = WaitForAnyBackendHit(start_index, start_index + 3);
-    size_t backend_index = absl::get<size_t>(result);
-    ASSERT_FALSE(absl::holds_alternative<Status>(result))
-        << absl::get<Status>(result).error_message();
-    saw_different_backend = (backend_index - start_index) > 0;
+    ASSERT_TRUE(result.ok()) << result.status();
+    size_t backend_index = *result;
+    if ((backend_index - start_index) > 0) {
+      return;
+    }
     // Toggle between backends 0-2 and 3-5
     start_index = 3 - start_index;
   }
-  EXPECT_TRUE(saw_different_backend);
+  FAIL() << "did not choose a different backend after 100 tries";
```

Comment:
Suggest removing the "after 100 tries" part, because it's likely to get out of sync with the actual number used in the loop above.

Reply:
Done